### PR TITLE
Add code sample for topic implementation [API-1522]

### PR DESCRIPTION
**Pull request by fatihozer0**
PR closed by Hazelcast automation as no activity (>3 months). Please reopen with comments, if necessary. Thank you for using Hazelcast and your valuable contributions

### CODE REVIEW

File: code_samples/topic.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    topic.addMessageListener((message) => {
+        console.log('Received message:\n', message);
+        // shut down the client once the message is received
+        client.shutdown().catch((err) => {
+            console.error('Failed to shut down the client:', err);
+        });
+    });
+
+    await topic.publish('Hello to distributed world');
+
```
Comment (srknzl) [inline code_samples/topic.js:33]:
```suggestion
```

File: code_samples/topic.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+} )().catch(err => {
```
Comment (srknzl) [inline code_samples/topic.js:34]:
```suggestion
})().catch(err => {
```

File: code_samples/topic.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ */
+'use strict';
+
+const { Client } = require('hazelcast-client');
+
+(async () => {
+    const client = await Client.newHazelcastClient();
+    const topic = await client.getTopic('my-distributed-topic');
+
+    topic.addMessageListener((message) => {
```
Comment (srknzl) [inline code_samples/topic.js:24]:
```suggestion
    await topic.addListener((message) => {
```

**Review by srknzl — CHANGES_REQUESTED** [review]
Seems fine when

File: code_samples/topic.js (hunk 1 above)
Comment (fatihozer0) [reply code_samples/topic.js:33]: Done

File: code_samples/topic.js (hunk 2 above)
Comment (fatihozer0) [reply code_samples/topic.js:34]: Done

File: code_samples/topic.js (hunk 3 above)
Comment (fatihozer0) [reply code_samples/topic.js:24]: And done.

**Comment by fatihozer0** [comment]
Should I update Reliable Topic sample since we deprecated addMessageListener method?

File: code_samples/topic.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+const { Client } = require('hazelcast-client');
+
+(async () => {
+    const client = await Client.newHazelcastClient();
+    const topic = await client.getTopic('my-distributed-topic');
+
+    topic.addListener((message) => {
+        console.log('Received message:\n', message);
+        // shut down the client once the message is received
```
Comment (srknzl) [inline code_samples/topic.js:26]: We can remove the listener here maybe.

**Pull request closed without merging**